### PR TITLE
[stable/jenkins] Cleanup agent.yamlTemplate rendering in XML config

### PR DESCRIPTION
**Pull request by wmcdona89**
#### What this PR does / why we need it:
Cleanup `agent.yamlTemplate` rendering in kubernetes pod template XML configuration.

This is a cosmetic change to render valid looking yaml. Technically, nothing was broken as the kubernetes plugin was able to parse the yaml despite the misalignment. 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
Given the following in stable/jenkins/agent-values.yaml
```yaml
agent:
  yamlTemplate: |-
    apiVersion: v1
    kind: Pod
    spec:
      tolerations:
      - key: "key"
        operator: "Equal"
        value: "value"
        effect: "NoSchedule"
```
#### original formatting
config.yaml
```yaml
<yaml>{{ tpl .Values.agent.yamlTemplate . | html | indent 4 | trim }}</yaml>
```
```
helm template stable/jenkins \
--show-only templates/config.yaml \
--values stable/jenkins/agent-values.yaml
```
```xml
              <yaml>apiVersion: v1
                kind: Pod
                spec:
                  tolerations:
                  - key: &#34;key&#34;
                    operator: &#34;Equal&#34;
                    value: &#34;value&#34;
                    effect: &#34;NoSchedule&#34;</yaml>
```
Jenkins screenshot
image

Pod YAML at runtime
```yaml
apiVersion: v1
kind: Podspec:
  tolerations:
  - effect: "NoSchedule"
    key: "key"
    operator: "Equal"
    value: "value"
```
#### proposed formatting
config.yaml
```yaml
  <yaml>
    {{- tpl (trim .Values.agent.yamlTemplate) . | html | nindent 4 }}
  </yaml>
```
```
helm template stable/jenkins \
--show-only templates/config.yaml \
--values stable/jenkins/agent-values.yaml
```
```xml
              <yaml>
                apiVersion: v1
                kind: Pod
                spec:
                  tolerations:
                  - key: &#34;key&#34;
                    operator: &#34;Equal&#34;
                    value: &#34;value&#34;
                    effect: &#34;NoSchedule&#34;
              </yaml>
```
Jenkins screenshot
image

Pod YAML at runtime
```yaml
apiVersion: v1
kind: Podspec:
  tolerations:
  - effect: "NoSchedule"
    key: "key"
    operator: "Equal"
    value: "value"
```
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/jenkins/templates/_helpers.tpl [hunk 1]
```diff
@@ -198,7 +198,7 @@ Returns kubernetes pod template configuration as code
 {{- end }}
   {{- if .Values.agent.yamlTemplate }}
   yaml: |-
-  {{- tpl ( trim .Values.agent.yamlTemplate ) . | nindent 4 }}
+  {{- tpl (trim .Values.agent.yamlTemplate) . | nindent 4 }}
```
Comment (wmcdona89) [inline stable/jenkins/templates/_helpers.tpl]: it appears the convention is to not include spaces inside parenthesis

**Review by torstenwalter — COMMENTED** [review]
@wmcdona89 What's left here to make it ready?

**Comment by wmcdona89** [comment]
Just finding the time to run a few more tests.

**Comment by wmcdona89** [comment]
I think I've convinced myself that the kubernetes plugin handles any extraneous whitespace, including a blank first line. It appears [org.csanchez.jenkins.plugins.kubernetes.PodTemplateUtils.parseFromYaml(String yaml)](https://javadoc.jenkins.io/plugin/kubernetes/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.html#parseFromYaml-java.lang.String-) is used to parse the yaml so I ran some unit tests for various inputs and every test passed. I think it's the underlying [Jackson library](https://github.com/FasterXML/jackson) that's handling the extraneous whitespace.

File: stable/jenkins/templates/_helpers.tpl [hunk 2]
```diff
@@ -220,10 +220,10 @@ Returns kubernetes pod template configuration as code
     {{- end }}
   {{- end }}
 {{- end }}
-  {{- if .Values.agent.yamlTemplate }}
+{{- if .Values.agent.yamlTemplate }}
   yaml: |-
-  {{- tpl ( trim .Values.agent.yamlTemplate ) . | nindent 4 }}
-  {{- end }}
+    {{- tpl (trim .Values.agent.yamlTemplate) . | nindent 4 }}
+{{- end }}
```
Comment (wmcdona89) [inline stable/jenkins/templates/_helpers.tpl:226]: updated indentation for readability

**Comment by wmcdona89** [comment]
@torstenwalter this is ready for a final review

**Review by torstenwalter — APPROVED** [review]
/lgtm

**Comment by torstenwalter** [comment]
As always nicely explained.

**Pull request merged**